### PR TITLE
Update IoTCentral to 2018-09-01

### PR DESCRIPTION
**Pull request by emgarten**
## Description

Swagger change: https://github.com/Azure/azure-rest-api-specs/pull/3466

* Update IoTCentral to 2018-09-01
* Update NuGet package version and assembly version to 1.0.0 stable for the GA release of IoT Central.
* Updated test data

---

This checklist is used to make sure that common guidelines for a pull request are followed.
- [x] Please add REST spec PR link to the SDK PR
- [x] **I have read the [contribution guidelines](https://github.com/Azure/azure-sdk-for-net/blob/AutoRest/.github/CONTRIBUTING.md).**
- [x] **The pull request does not introduce [breaking changes](https://github.com/Azure/azure-sdk-for-net/blob/AutoRest/Documentation/breaking-changes.md).**

### [General Guidelines](https://github.com/Azure/azure-sdk-for-net/blob/AutoRest/.github/CONTRIBUTING.md#general-guidelines)
- [x] Title of the pull request is clear and informative.
- [x] There are a small number of commits, each of which have an informative message. This means that previously merged commits do not appear in the history of the PR. For more information on cleaning up the commits in your PR, [see this page](https://github.com/Azure/azure-powershell/blob/dev/documentation/cleaning-up-commits.md).

### [Testing Guidelines](https://github.com/Azure/azure-sdk-for-net/blob/AutoRest/.github/CONTRIBUTING.md#testing-guidelines)
- [x] Pull request includes test coverage for the included changes.

### [SDK Generation Guidelines](https://github.com/Azure/azure-sdk-for-net/blob/AutoRest/.github/CONTRIBUTING.md#sdk-generation-guidelines)
- [x] If an SDK is being regenerated based on a new swagger spec, a link to the pull request containing these swagger spec changes has been included above.
- [x] The generate.cmd file for the SDK has been updated with the version of AutoRest, as well as the commitid of your swagger spec or link to the swagger spec, used to generate the code.
- [x] The `*.csproj` and `AssemblyInfo.cs` files have been updated with the new version of the SDK.

### CODE REVIEW

**Review by dsgouda — CHANGES_REQUESTED** [review]

File: src/SDKs/IotCentral/Management.IotCentral/Microsoft.Azure.Management.IotCentral.csproj [hunk 1]
```diff
@@ -8,7 +8,7 @@
     <Description>Provides management capabilities for Microsoft Azure IotCentral.</Description>
     <AssemblyTitle>Microsoft Azure IotCentral Management</AssemblyTitle>
     <AssemblyName>Microsoft.Azure.Management.IotCentral</AssemblyName>
-    <Version>0.9.0-preview</Version>
+    <Version>1.0.0</Version>
     <PackageTags>Microsoft Azure IotCentral;IotCentral management;IotCentral;</PackageTags>
     <PackageReleaseNotes>
       <![CDATA[
```
[the remark this answers is not in the document]
Comment (emgarten) [reply src/SDKs/IotCentral/Management.IotCentral/Microsoft.Azure.Management.IotCentral.csproj]: Good catch, I've updated this

File: src/SDKs/_metadata/iotcentral_resource-manager.txt [hunk 2]
```diff
@@ -1,33 +1,33 @@
 ﻿Installing AutoRest version: latest
 
-> autorest@2.0.4280 preinstall C:\Users\juste\AppData\Roaming\npm\node_modules\autorest
+> autorest@2.0.4283 preinstall C:\Users\juste\AppData\Roaming\npm\node_modules\autorest
```
[the remark this answers is not in the document]
Comment (emgarten) [reply src/SDKs/_metadata/iotcentral_resource-manager.txt]: I've updated everything, I don't see too many changes. I thought I was using the latest last time but I was getting some errors around `An exception occurred during a WebClient request`, that seems to be fixed today.

**Comment by emgarten** [comment]
🔔 updated

**Review by dsgouda — CHANGES_REQUESTED** [review]
Please replace `2017-07-01-privatepreview` with `2018-09-01` [here](https://github.com/Azure/azure-sdk-for-net/blob/b9208da181d1b874c82ba9d1d35f1b57375081bc/src/SDKs/IotCentral/AzSdk.RP.props#L4) and this should be good to merge

**Comment by emgarten** [comment]
@dsgouda fixed, thanks!

**Review by dsgouda — APPROVED** [review]
LGTM

**Pull request merged**